src/models/Settings.js

```diff
@@ -352,9 +353,24 @@ export default Model.extend({
   },
 
   initialize: function(options) {
-    if (!options.baseUrl) {
+    let { baseUrl, colors } = options;
+    if (!baseUrl) {
+      // infer baseUrl from the issuer
+      const { authClient } = options;
+      if (authClient) {
+        baseUrl = authClient.getIssuerOrigin();
+      } else {
+        // issuer can be passed at top-level or in authParams
+        let { issuer, authParams } = options;
+        issuer = issuer || authParams?.issuer;
+        baseUrl = issuer?.split('/oauth2/')[0];
+      }
+      this.set('baseUrl', baseUrl);
+    }
+
+    if (!baseUrl) {
```

Comment:
this can be merged to the `if` above right ?

Reply:
It could... but the block above sets the baseUrl if it is not set, inferring value from issuer. I thought this was a distinct logical block from throwing the error.

Reply:
I was thinking "setting defaults" vs. "asserting required", but maybe it does make more sense to do it by parameter name.